ffmpeg/lpms_ffmpeg.c

```diff
@@ -753,6 +797,37 @@ static int open_video_decoder(input_params *params, struct input_ctx *ctx)
       AVCodec *c = avcodec_find_decoder_by_name("h264_cuvid");
       if (c) codec = c;
       else fprintf(stderr, "Cuvid decoder not found; defaulting to software\n");
+    } else if (AV_CODEC_ID_H264 == codec->id &&
+                       AV_HWDEVICE_TYPE_VAAPI == params->hw_type) {
+      enum AVHWDeviceType type;
+      type = av_hwdevice_find_type_by_name("vaapi");
+      if (type == AV_HWDEVICE_TYPE_NONE) {
+        fprintf(stderr, "Device type vaapi is not supported.\n");
+        fprintf(stderr, "Available device types:");
+        while((type = av_hwdevice_iterate_types(type)) != AV_HWDEVICE_TYPE_NONE)
+          fprintf(stderr, " %s", av_hwdevice_get_type_name(type));
+        fprintf(stderr, "\n");
+        ret = -1;
+        goto open_decoder_err;
+      }
+
+      AVCodec *c = NULL;
+      int i;
+      ret = av_find_best_stream(ic, AVMEDIA_TYPE_VIDEO, -1, -1, &c, 0);
+      if (c) {
+        codec = c;
+        for (i = 0;; i++) {
+          const AVCodecHWConfig *config = avcodec_get_hw_config(codec, i);
```

Comment:
Curious, is there a reason this approach was taken over the example of `open_input_file` in https://git.ffmpeg.org/gitweb/ffmpeg.git/blob_plain/HEAD:/doc/examples/vaapi_transcode.c ? Were there issues with running on AMD?

Reply:
On some devices like yours, nvidia gpu are listed under `/dev/dri/renderD*`. This check makes sure to inform user that VAAPI is not supported for Nvidia GPU